### PR TITLE
EZP-28592: Update the UDW look according to designs

### PR DESCRIPTION
**Pull request by sunpietro**
https://jira.ez.no/browse/EZP-28592

**The Browse tab**
zrzut ekranu 2017-12-19 o 19 41 49

**The Search tab**
zrzut ekranu 2017-12-19 o 19 42 47

The Search tab looks differently in the end. There are 2 features missing. The X button in the search input and the new pagination. These will be implemented in the future PRs when the features will be specified.

### CODE REVIEW

File: src/modules/universal-discovery/components/selected-content/css/selected.content.popup.component.css [hunk 1]
```diff
@@ -1,12 +1,28 @@
 .c-selected-content-popup .c-popup {
-    width: 640px;
+    width: fit-content;
```
Comment (dew326) [inline src/modules/universal-discovery/components/selected-content/css/selected.content.popup.component.css]: According to caniuse.com Edge doesn't support fit-content

File: src/modules/universal-discovery/components/selected-content/selected.content.item.component.js [hunk 2]
```diff
@@ -3,16 +3,29 @@ import PropTypes from 'prop-types';
...
 
     return (
         <div className="c-selected-content-item">
-            <div className="c-selected-content-item__remove" onClick={() => onRemove(data.id)}>&times;</div>
             <div className="c-selected-content-item__wrapper">
                 <div className="c-selected-content-item__name">{data.ContentInfo.Content.Name}</div>
-                <div className="c-selected-content-item__type">{contentTypeName}</div>
+                <div className="c-selected-content-item__type">{icon} {contentTypeName}</div>
+            </div>
+            <div className="c-selected-content-item__remove" onClick={() => onRemove(data.id)}>
```
Comment (dew326) [inline src/modules/universal-discovery/components/selected-content/selected.content.item.component.js:25]: I think this should work `onClick={onRemove.bind(this, data.id)}`

File: src/modules/universal-discovery/css/universal.discovery.module.css [hunk 3]
```diff
@@ -78,33 +91,40 @@
...
 .m-ud__action--confirm {
-    padding: 8px;
-    font-size: 16px;
     font-weight: 700;
-    border-radius: 4px;
-    background: linear-gradient(to bottom, #fff 0%, #e6e6e6 100%);
-    border: 1px solid #bbb;
-    cursor: pointer;
-    transition: background .3s var(--transition-function);
+    background: #F15A10;
```
Comment (dew326) [inline src/modules/universal-discovery/css/universal.discovery.module.css]: lowercase?

**Review by dew326 — CHANGES_REQUESTED** [review]

File: src/modules/universal-discovery/components/selected-content/css/selected.content.popup.component.css (hunk 1 above)
Comment (lserwatka) [reply src/modules/universal-discovery/components/selected-content/css/selected.content.popup.component.css]: we support Edge

File: src/modules/universal-discovery/components/content-meta-preview/css/content.meta.preview.component.css [hunk 4]
```diff
@@ -1,55 +1,106 @@
...
+    font-size: 15px;
+    color: #515151;
+    line-height: 20px;
+    display: flex;
+    align-items: center;
+}
+
+.c-meta-preview__image-wrapper {
+    width: 190px;
+    height: 190px;
```
Comment (inakijv) [inline src/modules/universal-discovery/components/content-meta-preview/css/content.meta.preview.component.css:41]: With `height: 210px;` and `width: 210px` everything would be more aligned within it

File: src/modules/universal-discovery/components/content-meta-preview/css/content.meta.preview.component.css [hunk 5]
```diff
@@ -1,55 +1,106 @@
...
+    font-size: 16px;
     font-weight: 700;
+    line-height: 20px;
     margin: 8px 0;
+    word-break: break-all;
 }
 
-.c-meta-preview__type {
+.c-meta-preview__content-type {
     margin: 8px 0;
```
Comment (inakijv) [inline src/modules/universal-discovery/components/content-meta-preview/css/content.meta.preview.component.css]: I would suggest to change it for `margin-bottom: 4px;` only.

File: src/modules/universal-discovery/components/selected-content/css/selected.content.popup.component.css [hunk 6]
```diff
@@ -1,12 +1,28 @@
...
     flex-wrap: wrap;
 }
 
+.c-selected-content-popup .c-popup__header {
+    margin-bottom: 8px;
+}
+
+.c-selected-content-popup .c-popup__title {
+    color: #2b84b1;
+    font-size: 15px;
```
Comment (inakijv) [inline src/modules/universal-discovery/components/selected-content/css/selected.content.popup.component.css]: Please change it to `font-size: 17px` for font hierarchy purposes.

File: src/modules/universal-discovery/components/selected-content/css/selected.content.popup.component.css [hunk 7]
```diff
@@ -1,12 +1,28 @@
...
+    line-height: 20px;
+}
+
+.c-selected-content-popup .c-popup__close {
+    right: 0;
+}
+
 .c-selected-content-popup .c-selected-content-item {
     flex: 1 1 calc(50% - 8px);
     max-width: calc(50% - 8px);
```
Comment (inakijv) [inline src/modules/universal-discovery/components/selected-content/css/selected.content.popup.component.css:28]: Please add `border-radius: 2px 0 0 2px;` for design purposes.

File: src/modules/universal-discovery/components/selected-content/css/selected.content.item.component.css [hunk 8]
```diff
@@ -5,20 +5,27 @@
...
+    background: #b3071c;
+}
+
+.c-selected-content-item__remove .ez-icon {
+    fill: #fff;
+    width: 12px;
+    height: 12px;
 }
 
 .c-selected-content-item__wrapper {
```
Comment (inakijv) [inline src/modules/universal-discovery/components/selected-content/css/selected.content.item.component.css:32]: Please add within `.c-selected-content-item__wrapper` a `padding: 4px 6px;` for alignment style.

File: src/modules/universal-discovery/components/selected-content/css/selected.content.item.component.css [hunk 9]
```diff
@@ -5,20 +5,27 @@
...
+    flex: 0 0 20px;
     display: flex;
     justify-content: center;
     flex-direction: column;
     text-align: center;
+    align-items: center;
     font-weight: 700;
     cursor: pointer;
-    transition: background .3s var(--transition-function);
+    transition: background .2s cubic-bezier(.25, .8, .25, 1);
```
Comment (inakijv) [inline src/modules/universal-discovery/components/selected-content/css/selected.content.item.component.css]: Please add `border-radius: 0 2px 2px 0;` for styling purposes.

File: src/modules/universal-discovery/components/selected-content/css/selected.content.item.component.css [hunk 10]
```diff
@@ -29,4 +36,24 @@
 .c-selected-content-item__name {
     margin-bottom: 16px;
     font-weight: 700;
+    word-break: break-all;
+    font-size: 13px;
+    line-height: 15px;
```
Comment (inakijv) [inline src/modules/universal-discovery/components/selected-content/css/selected.content.item.component.css]: After testing the design I would suggest:
- to remove `line-height: 15px;`
- change to `font-size: 15px;`
- add `padding: 0 4px;`

File: src/modules/universal-discovery/components/selected-content/css/selected.content.item.component.css [hunk 11]
```diff
@@ -29,4 +36,24 @@
...
+}
+
+.c-selected-content-item__type {
+    font-size: 13px;
+    line-height: 15px;
+    word-break: break-all;
+    color: #878787;
+    background: #f3f3f3;
+    display: flex;
+    align-items: center;
```
Comment (inakijv) [inline src/modules/universal-discovery/components/selected-content/css/selected.content.item.component.css]: Please change:
- `color: #555;` for color contrast purposes.
- remove `line-height: 15px;`
- add `padding: 2px 4px;`
- add `border-radius: 2px;`

File: src/modules/common/popup/css/popup.component.css [hunk 12]
```diff
@@ -24,3 +24,9 @@
     font-size: 32px;
     cursor: pointer;
 }
+
```
Comment (inakijv) [inline src/modules/common/popup/css/popup.component.css:27]: Please add to `.c-selected-content-popup .c-popup__header` for styling purposes:
- `margin-bottom: 4px;`
- `margin-left: 6px;`

File: src/modules/universal-discovery/css/universal.discovery.module.css [hunk 13]
```diff
@@ -78,33 +91,40 @@
...
 .m-ud__action--confirm:hover,
 .m-ud__action--confirm:focus {
-    background: linear-gradient(to bottom, #e6e6e6 0%, #fff 100%);
+    background: #cb3400;
 }
 
 .m-ud__action--confirm[disabled] {
     cursor: not-allowed;
     background: #e6e6e6;
+    color: #333;
```
Comment (inakijv) [inline src/modules/universal-discovery/css/universal.discovery.module.css]: Please change disabled state for consistency purposes:
- remove `background:#e6e6e6;`
- remove `color: #333;`
- add `opacity: .3;`

File: src/modules/universal-discovery/components/tab-nav/css/tab.nav.item.component.css [hunk 14]
```diff
@@ -1,21 +1,35 @@
...
+    font-weight: 700;
+    border-radius: 4px 4px 0 0;
 }
 
-.c-tab-nav-item--selected {
-    background: #fff;
-    border-top-color: #b30;
+.c-tab-nav-item__icon {
+    width: 16px;
+    height: 16px;
```
Comment (inakijv) [inline src/modules/universal-discovery/components/tab-nav/css/tab.nav.item.component.css]: Please increase size for the icon:
- `width: 18px;`
- `height: 18px;`

File: src/modules/universal-discovery/components/tab-nav/css/tab.nav.item.component.css [hunk 15]
```diff
@@ -1,21 +1,35 @@
 .c-tab-nav-item {
-    background: #f0fafa;
+    background: #eee;
     border: 0 none;
-    border-top: 3px solid #f0fafa;
     font-size: 16px;
-    padding: 8px 16px;
+    line-height: 30px;
+    padding: 0 16px;
```
Comment (inakijv) [inline src/modules/universal-discovery/components/tab-nav/css/tab.nav.item.component.css]: Could you please change it to:
- `padding: 4px 16px;`

File: src/modules/universal-discovery/components/search/css/search.results.item.component.css [hunk 16]
```diff
@@ -21,18 +23,26 @@
...
-.c-search-results-item__btn--preview:hover,
 .c-search-results-item__btn--preview:focus {
-    background: linear-gradient(to bottom, #e6e6e6 0%, #fff 100%);
+    outline: none;
+}
+
+.c-search-results-item__btn--preview .ez-icon {
+    fill: #445A64;
+    width: 32px;
+    height: 32px;
```
Comment (inakijv) [inline src/modules/universal-discovery/components/search/css/search.results.item.component.css]: Could you please change them to:
- `fill: #0f6d95;`
- `width: 30px;`
- `height: 30px;`

File: src/modules/universal-discovery/components/search/css/search.component.css [hunk 17]
```diff
@@ -9,4 +10,37 @@
...
+    display: flex;
+}
+
+.c-search__input {
+    border: 2px solid #2b84b1;
+    border-right: 0 none;
+    border-radius: 4px 0 0 4px;
+    background-color: #fff;
+    line-height: 30px;
+    padding: 0 8px;
```
Comment (inakijv) [inline src/modules/universal-discovery/components/search/css/search.component.css:22]: Please change:
- border to `border: 1px solid rgba(51,51,51,.15);`
- add a `:focus` state with `border-color: #106d95;`

Also, Do you believe it's possible to add a `min-width: 300px;`, input search box should be a little more wider.

File: src/modules/universal-discovery/css/universal.discovery.module.css [hunk 18]
```diff
@@ -12,11 +12,8 @@
     left: 50%;
     transform: translate(-50%, -50%);
     background: #eee;
-    padding: 16px;
     box-shadow: 0 0 30px 0 rgba(0,0,0,.35);
-    display: grid;
-    grid-template-columns: auto;
-    grid-template-rows: 48px 48px 1fr 48px;
+    border-radius: 6px;
```
Comment (inakijv) [inline src/modules/universal-discovery/css/universal.discovery.module.css]: Please change: 
- `border-radius: 4px;`
- I would also suggest to change, at least, width to `width: 95vw;`, and also consider to change height to `height: 95vh;` in order to maximize UDW's features when in medium screen devices especially.

File: src/modules/universal-discovery/css/universal.discovery.module.css [hunk 19]
```diff
@@ -45,12 +59,12 @@
 }
 
 .m-ud__content {
-    grid-row: 3 / 6;
+    grid-area: main;
+    padding: 16px;
+    background: #e3e3e3;
```
Comment (inakijv) [inline src/modules/universal-discovery/css/universal.discovery.module.css:64]: Please add:
- `border-radius: 4px;`

File: src/modules/universal-discovery/components/content-meta-preview/css/content.meta.preview.component.css [hunk 20]
```diff
@@ -1,55 +1,106 @@
 .c-meta-preview {
     background: #fff;
     height: 100%;
-    padding: 16px;
+    overflow-y: auto;
+    padding: 10px;
```
Comment (inakijv) [inline src/modules/universal-discovery/components/content-meta-preview/css/content.meta.preview.component.css:5]: Please add:
`border-radius: 4px;`

**Review by inakijv — CHANGES_REQUESTED** [review]
Hello @sunpietro, thank you so much for this contribution. It looks great. I am adding small comments regarding styling. Thank you!

**Comment by inakijv** [comment]
If possible, please also add three more small additions:
- To `.c-finder-tree-branch` add `border-radius: 4px;`
- To `.c-finder__branches` add `border-radius: 4px;`
- To `.c-search-panel` add `border-radius: 4px;`

Thank you!

File: src/modules/universal-discovery/components/selected-content/selected.content.item.component.js (hunk 2 above)
Comment (sunpietro) [reply src/modules/universal-discovery/components/selected-content/selected.content.item.component.js:25]: I don't want to override context. That the reason the code looks like this.

**Comment by sunpietro** [comment]
ping @dew326

**Review by dew326 — APPROVED** [review]

**Comment by lserwatka** [comment]
@micszo has anyone tested this PR?

**Comment by micszo** [comment]
@mnocon still is.

**Comment by sunpietro** [comment]
@lserwatka there are 2 minor visual issues reported by @mnocon 
I'll fix them asap.

**Pull request merged**